creator-node/src/routes/tracks.js

```diff
@@ -16,72 +16,93 @@ module.exports = function (app) {
    * @dev - currently stores each segment twice, once under random file UUID & once under IPFS multihash
    *      - this should be addressed eventually
    */
-  app.post('/track_content', authMiddleware, ensurePrimaryMiddleware, syncLockMiddleware, trackFileUpload.single('file'), handleResponse(async (req, res) => {
-    if (req.fileFilterError) return errorResponseBadRequest(req.fileFilterError)
-    const routeTimeStart = Date.now()
-    let codeBlockTimeStart = Date.now()
+  app.post('/track_content',
+    authMiddleware,
+    syncLockMiddleware,
+    trackFileUpload.fields([
+      { name: 'file', maxCount: 1 },
+      { name: 'fileSegments' }
+    ]),
+    handleResponse(async (req, res) => {
+      if (req.fileFilterError) return errorResponseBadRequest(req.fileFilterError)
+      const routeTimeStart = Date.now()
+      let codeBlockTimeStart = Date.now()
+      const { file: trackFiles, fileSegments } = req.files
+      const trackFile = trackFiles[0]
+      const trackName = trackFile.originalname
+      const dirName = path.dirname(trackFile.path)
+      let segmentFilePaths
+      let manifestFile
+      if (!fileSegments) {
+      // create and save track file segments to disk
+        try {
+          req.logger.info(`Segmenting file ${trackName}...`)
+          const ffmpegSegmentFiles = await ffmpeg.segmentFile(req, dirName, trackFile.filename, trackName)
+          segmentFilePaths = ffmpegSegmentFiles.map(file => ({
+            path: path.resolve(dirName, 'segments', file),
+            originalname: file
+          }))
+          manifestFile = path.resolve(dirName, trackName.split('.')[0] + '.m3u8')
+          req.logger.info(`Time taken in /track_content to segment track file: ${Date.now() - codeBlockTimeStart}ms for file ${trackName}`)
+        } catch (err) {
+          removeTrackFolder(req, dirName)
+          return errorResponseServerError(err)
+        }
+      } else {
+        manifestFile = fileSegments.find(file => file.originalname.endsWith('.m3u8')).path
+        segmentFilePaths = fileSegments.filter(file => file.originalname.startsWith('segment'))
```

Comment:
is there a way to cheaply confirm that these segments are indeed the ones generated from the master...? maybe a duration check or something simple like that? only if it doesn't require an additional ffprobe op here would this be feasible - i know that m3u8 files don't contain info about the master but it might be relatively cheap client side